charts/tidb-cluster/templates/monitor.yaml

```diff
@@ -0,0 +1,103 @@
+{{- if .Values.monitor.create }}
+apiVersion: pingcap.com/v1alpha1
+kind: TidbMonitor
+metadata:
+  # for backward compatibility, name should not be {{ template "cluster.name" . }}-monitor
+  # or the names of the pods/svcs would be changed, like tidb-cluster-grafana -> tidb-cluster-monitor-grafana
```

Comment:
For backward compatibility, I think this problem should be resolved in `TidbMonitor` Controller. I would create an new issue to solve this problem.